### ISSUE TITLE
Issues after Collection Structure stuff

### ISSUE THREAD

**Issue by damcclean** [doublethreedigital/duplicator#14]
A user reported an issue where whenever they tried to duplicate an entry, they'd get `Undefined offset: 0`. I talked with the user and they shared their collection's yaml file.

```yaml
title: Pages
revisions: false
route: '{parent_uri}/{slug}'
structure:
  root: true
  tree:
    -
      entry: home
    -
      entry: 3f9e0dbd-9b99-42ac-b1af-4bced9f8c513
    -
      entry: e6151e9b-170c-4415-8c81-1cb6f5e26bd0
    -
      entry: 569f0065-37f9-4692-8c0a-1cb2dd4a68cf
    -
      entry: 867c58a1-d4c5-4ee7-9189-46f1dfad12b0
    -
      entry: cd93e21c-106e-40f8-a25b-dc34b61c0fee
```

## Steps to reproduce

1. With a structure, similar to the one above, attempt to duplicate an entry.
2. Be presented with the error.

## Expected behaviour

I expected the entry to be duplicated.

## Environment information

**Statamic version:** 3.0.32

**Comment by mikemartin** [comment]
Hey @damcclean 
I've hit the same issue with this collection structure:
```
title: Pages
template: default
layout: layout
revisions: true
route: '{parent_uri}/{slug}'
sort_dir: asc
structure:
  root: true
  tree:
    -
      entry: home
    -
      entry: dde41cc8-b657-42c8-98d1-cc67bf1f5022
      children:
        -
          entry: 510cbbd9-e2ae-4359-b97a-376a302bbd6f
        -
          entry: fa89a1aa-1e79-42bc-9960-a50169939e00
    -
      entry: 117bfda5-c05e-4749-8788-f75f0a1fcfbf
    -
      entry: a36f09b3-a4b2-4dc1-8201-bfe37c187d98
    -
      entry: 7fa79475-41c0-46aa-893f-040037ba3ba0
    -
      entry: d7d3b285-2a7e-4edd-9ea2-8ed9d9213be8
    -
      entry: c75e7fa3-a57f-4be4-8e06-6369af576353
    -
      entry: 56d70357-3602-4592-a41d-1f27db1f9b6a
      children:
        -
          entry: 38914e71-2717-4475-ac74-83dd5b7905cd
        -
          entry: 8f681f24-e2a9-49de-84ee-29614c6bb52f
    -
      entry: cb1e412b-2aa9-40c8-969b-bd5408532307
    -
      entry: 42998498-f271-4210-8b0c-03d8f257cccc
    -
      entry: 264ed03a-b45f-4a2d-bf18-bdc398f091ae
    -
      entry: 4631cfb1-550c-4932-b7b2-6f7b1bb8910a
    -
      entry: 13f2533a-ee47-4817-845a-9e6692ee6634
    -
      entry: db87b28f-2d6a-43be-86d1-09fac3be36aa
    -
      entry: d3a8eb5c-e5a6-4171-b043-9fca5c1cb11d
    -
      entry: 52285f1c-f97f-47d9-9cb0-e0e935a339ef
    -
      entry: 5b71cbd3-6ea9-4307-925c-9cd69c9b4d3a
```

**Comment by mikemartin** [comment]
Hey @damcclean 
Quick follow up. I've removed all pages from the collection leave only 2 entries. 'Home' and another page. 

I can duplicate the homepage but the other page results in the offset error.

Pages.yaml:

```
title: Pages
template: default
layout: layout
revisions: true
route: '{parent_uri}/{slug}'
sort_dir: asc
structure:
  root: true
  tree:
    -
      entry: home
    -
      entry: 846d504a-517d-4974-9715-34d32233312b
```

Sleeping-tips.md content (results in error when duplocated)
```
---
title: 'Download Your Sleeping Tips PDF Now!'
id: 846d504a-517d-4974-9715-34d32233312b
---
```

**Comment by damcclean** [comment]
Thanks for bringing this back up! I almost forgot entirely about this issue. I'll see to it today.

**Comment by damcclean** [comment]
This issue should be fixed in [v1.2.2](https://github.com/doublethreedigital/duplicator/releases/tag/v1.2.2).
